pokemongo_bot/walkers/step_walker.py

```diff
@@ -96,20 +96,23 @@ def _calc_bearing(self, start_lat, start_lng, dest_lat, dest_lng):
     def get_next_position(self, origin_lat, origin_lng, origin_alt, dest_lat, dest_lng, dest_alt, speed):
         origin = Point(origin_lat, origin_lng, origin_alt)
 
+        if speed == 0:
+            speed = self.precision
+
         total_distance = distance(origin_lat, origin_lng, dest_lat, dest_lng)
 
+        if total_distance == 0:
+            total_distance = self.precision
+
         if speed == float("inf"):
             dist = total_distance
         else:
             dist = min(total_distance, speed)
 
-        if dist == 0:
-            dist = self.precision
-
         bearing = self._calc_bearing(origin_lat, origin_lng, dest_lat, dest_lng)
         offset_angle = (100 / dist) * (self.precision / 1.74)
 
         next_lat, next_lng, _ = VincentyDistance(kilometers=dist * 1e-3).destination(origin, bearing + uniform(-offset_angle, offset_angle))
-        next_alt = dest_alt + random_alt_delta()
+        next_alt = origin_lat + (dist / total_distance) * (dest_alt - origin_lat) + random_alt_delta()
```

Comment:
if total_distance == 0

you have a DivisionByZero exception in:

dist / total_distance